test/e2e/autoscale_test.go

```diff
@@ -126,6 +128,60 @@ func generateTraffic(ctx *testContext, concurrency int, duration time.Duration,
 	return nil
 }
 
+func generateTrafficAtFixedRPS(ctx *testContext, rps float64, duration time.Duration, stopChan chan struct{}) error {
+	var (
+		totalRequests      int32
+		successfulRequests int32
+	)
+
+	ctx.t.Logf("Maintaining %v RPS requests for %v.", rps, duration)
+	client, err := pkgTest.NewSpoofingClient(ctx.clients.KubeClient, ctx.t.Logf, ctx.domain, test.ServingFlags.ResolvableDomain)
+	if err != nil {
+		return fmt.Errorf("error creating spoofing client: %v", err)
+	}
+
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s?sleep=100", ctx.domain), nil)
+	if err != nil {
+		return fmt.Errorf("error creating HTTP request: %v", err)
+	}
+
+	ticker := time.NewTicker(time.Second / time.Duration(rps))
```

Comment:
Can you just use vegeta? It will give you the stats as well.

Reply:
Done.